components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java

```diff
@@ -80,6 +80,52 @@ private ApplicationConstants() {
     public static final int DEFAULT_FETCH_CHUNK_SIZE = 50;
     public static final String FETCH_CHUNK_SIZE = "FetchChunkSize";
 
+    // Named query fields
+    public static final String OFFSET = "OFFSET";
+    public static final String LIMIT = "LIMIT";
+    public static final String ZERO_BASED_START_INDEX = "ZERO_BASED_START_INDEX";
+    public static final String ONE_BASED_START_INDEX = "ONE_BASED_START_INDEX";
+    public static final String END_INDEX = "END_INDEX";
+
+    /**
+     * Enum for application service related errors in the format of
+     * Error Code - code to identify the error
+     * Error Message - What went wrong
+     */
+    public enum ErrorMessage {
+
+        ERROR_INVALID_OFFSET("60001", "Invalid offset requested. Offset value should be zero or greater than zero."),
```

Comment:
Since this a user api, shouldn't we use 1xxxxx , 15xxxx range?

Reply:
From the OSGI service I don't think we can differentiate user vs server APIs.
That differentiation comes with user personas interacting with REST or Soap APIs that are exposed.
Internally, a REST API invoked by end users as well as privileged/system users can use the same underlying OSGI service.
Therefore, I would treat OSGI services as server APIs and we need to have a range specified for OSGI services to use